python/lsst/dax/imgserv/api_soda.py

```diff
@@ -124,45 +143,272 @@ def img_capabilities():
 
 @image_soda.route("/examples", methods=["GET"])
 def img_examples():
+    """ Get /examples for the service. """
     html = current_app.soda.get_examples(_getparams())
     return make_response(html)
 
 
 @image_soda.route("/tables", methods=["GET"])
 def img_tables():
+    """ Get /tables for the service. """
     xml = current_app.soda.get_tables(_getparams())
     return make_response(xml)
 
 
 @image_soda.route("/sia", methods=["GET"])
 def img_sia():
+    """" Get the /sia service endpoint."""
     resp = current_app.soda.do_sia(_getparams())
     return make_response(resp)
 
 
 @image_soda.route("/sync", methods=["GET", "PUT", "POST"])
 def img_sync():
+    """ Service the /sync request.
+
+    Returns
+    -------
+    resp : `flask.Response`
+        the response.
+    """
     if not request.args:
         soda_url = url_for('api_image_soda.img_sync', _external=True)
         # no parameters present, return service status
         return _service_response(soda_url)
     image = current_app.soda.do_sync(_getparams())
     if image:
-        return _data_response(image)
+        with tempfile.NamedTemporaryFile(prefix="img_", suffix=".fits") as fp:
+            image.writeFits(fp.name)
+            resp = send_file(fp.name,
+                             mimetype="image/fits",
+                             as_attachment=True,
+                             attachment_filename=os.path.basename(fp.name))
+            return resp
```

Comment:
Are you writing the file to disk so that you can use this function?  You should probably use some kind of in memory object to not have to write it to disk just to throw it away (also, you'd have to make sure you have enough temp space here)

Reply:
Yes, that's correct, afw only has this writeFits() function. In the long run, since we can expect async workers to process most if not all of the requests, disk-based operation will be just fine.

Reply:
I still think we should do this in memory, for example BytesIO is a file like object, but just writes to memory.  I take it writeFits takes a file like python object, so you should be able to pass this in.

For async, I think it will be different, since async workers will have to push the resulting file to an object store or some 3rd place where it can be accessed.  (Otherwise, the client has to somehow find the worker that has the result and have it served from there, which is really kind of nasty if you try to redeploy or scale down workers).

Reply:
`writeFits` can write to an in memory FITS file.

Reply:
Prefacing this comment with the fact that I haven't looked at how this is deployed, so it may be moot:

An in memory file may be good, but if it is multiprocess deployment, which I think it is because it's `sync`, you will need to be careful to understand the deployment and the the largest FITS file you may possibly write and make sure that your resources/processes are allocated appropriately. If 20 sync jobs are being simultaneously processed with 20 processes writing out 2GB FITS files and you only have 32GB for the deployment you are going to get your deployment killed in kubernetes. If the worker deployment is a single process, you also need to size the max memory appropriately large (and also be extra careful to ensure zero-copy)

Reply:
and given the conversation on Slack middleware just now it does make me wonder whether all of this could be a `butler.put` that could be configured to go to S3 or file system...

Reply:
which is to say - I'd probably try to set a maximum image size of 2GB and allocate only 8 processes for 32GB deployment if using writeFits with a memory file.

Reply:
With this PR (once merged), the 'backend' component of imgserv can definitely use butler.put for storing the results, particularly of large size. On in-memory vs file-based, I'm trying to dig up the old/original implementation which was memory-based....

Reply:
Looking at the blame history of this code: https://github.com/lsst/dax_imgserv/commit/b7e61cd3c4fdaadbf6d99a7e6678eae7b303dc4d
If you look at the since deleted function  _data_response(image), that's what we want to bring back here?  As we all know, we'll be moving sync to be based on /async, which will be file-based or butler.put.  Question: what to do in the meantime?

Reply:
Just responding to the running out of memory, that's totally true and a real worry.  Same for running out of disk if we write it to disk.  For celery, it's easy to configure the number of running jobs at a time since it tracks each job's results and status independently.  (I'm not sure if we're going to have multiple cutouts being tracked as one job, but it seems better to expand out the multi-cutouts first, then dispatch each one as its own celery job).  For sync, you're right in that you don't know how many people might be calling it simultaneously

Reply:
As reminder, imgserv does have a open ticket : DM-20495 for setting appropriate resource limits for deployment.